### PR TITLE
SFINT-3009 Fix an issue where the text is not changed when clicked

### PR DESCRIPTION
**Pull request by jeremierobert-coveo**
* Added UTs.
* Added an option to set the tooltip success label.
* Ensure the success label is shown on success.

(This PR is therefore called shame)

### CODE REVIEW

File: src/components/CopyToClipboard/CopyToClipboard.ts [hunk 1]
```diff
@@ -72,14 +79,21 @@ export class CopyToClipboard extends ResultAction {
...
                 await navigator.clipboard.writeText(text);
-                this.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText = l('CopyToClipboard_copied');
             } catch (err) {
                 this.logger.error('Copy to clipboard failed.', text, err);
                 this.copyToClipboardFallback(text);
             }
         } else {
             this.copyToClipboardFallback(text);
         }
+        this.setToolipText(this.options.successTooltip);
```
Comment (louis-bompart) [inline src/components/CopyToClipboard/CopyToClipboard.ts]: Would remove the parameter, it's stored in `this.options`

File: src/components/CopyToClipboard/CopyToClipboard.ts [hunk 2]
```diff
@@ -72,14 +79,21 @@ export class CopyToClipboard extends ResultAction {
...
+    }
+
+    private setToolipText(text: string) {
+        const tooltipElement = this.element.querySelector<HTMLElement>('.coveo-caption-for-icon');
+        if (tooltipElement && text) {
```
Comment (louis-bompart) [inline src/components/CopyToClipboard/CopyToClipboard.ts]: What should be the behavior if I set `this.options.successTooltip=''` ?

File: src/components/CopyToClipboard/CopyToClipboard.ts [hunk 3]
```diff
@@ -16,6 +16,11 @@ export interface ICopyToClipboardOptions extends IResultActionOptions {
      */
     tooltip?: string;
 
+    /**
+     * The tooltip that displays when the action succeed.
```
Comment (lbergeron) [inline src/components/CopyToClipboard/CopyToClipboard.ts]: `succeed.` --> `succeeds`

File: src/components/CopyToClipboard/CopyToClipboard.ts (hunk 2 above)
Comment (mikegron) [reply src/components/CopyToClipboard/CopyToClipboard.ts]: Je dirais de ne pas changer le texte

Comment (jeremierobert-coveo) [reply src/components/CopyToClipboard/CopyToClipboard.ts]: Exact

File: tests/components/CopyPaste/CopyPaste.spec.ts [hunk 4]
```diff
@@ -62,6 +70,11 @@ describe('CopyToClipboard ResultAction', () => {
                 testComponent.element.click();
                 expect(execCommandStub.called).toBeTrue();
             });
+
+            it('should change the tooltip text for "Copied!"', async () => {
```
Comment (lbergeron) [inline tests/components/CopyPaste/CopyPaste.spec.ts]: You should add tests to validate tooltip text when using custom `options.tooltip` and `options.successTooltip`.

File: src/components/CopyToClipboard/CopyToClipboard.ts (hunk 2 above)
Comment (louis-bompart) [reply src/components/CopyToClipboard/CopyToClipboard.ts]: We need a test for that then :D

File: src/components/CopyToClipboard/CopyToClipboard.ts (hunk 1 above)
Comment (jeremierobert-coveo) [reply src/components/CopyToClipboard/CopyToClipboard.ts]: Nope the function is use twice with different param

File: tests/components/CopyPaste/CopyPaste.spec.ts [hunk 5]
```diff
@@ -71,17 +87,59 @@ describe('CopyToClipboard ResultAction', () => {
...
+            it('should use the provided successTooltip', async () => {
+                const tooltipText = 'some success tooltip';
+                buildResultAction({ successTooltip: tooltipText });
+                testComponent.element.click();
+
+                await Promise.resolve();
+                expect(testComponent.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText).toEqual(tooltipText);
+            });
+
+            it('should not change the tooltip if the successTooltip is empty-like', async () => {
```
Comment (lbergeron) [inline tests/components/CopyPaste/CopyPaste.spec.ts]: Why not `...if the successTooltip is empty.`? Using `empty-like` hints me that you should have tested `undefined` as well.

Comment (jeremierobert-coveo) [reply tests/components/CopyPaste/CopyPaste.spec.ts]: null undefined or '' is the same

Comment (lbergeron) [reply tests/components/CopyPaste/CopyPaste.spec.ts]: I know but your test is checking only one of those 3 cases right now. The best would be to validate those 3 cases. If we stick to validating only the empty string, I would rename the test case to make it clear.

Comment (jeremierobert-coveo) [reply tests/components/CopyPaste/CopyPaste.spec.ts]: oki

**Pull request merged**